x-pack/plugins/alerting/common/rule.ts

```diff
@@ -89,8 +90,23 @@ export interface RuleAction {
   };
 }
 
-export interface RuleAggregations {
-  alertExecutionStatus: { [status: string]: number };
+type IndexType = Record<string, unknown>;
```

Comment:
I will be honest I don't know what the purpose of a lot of these properties is either since it's before my time, but you are right in that we should start documenting this. Feel free to create a ticket if you would like to see some traction in that area.